src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -377,6 +377,10 @@ private static function resolveUrl(array $url, ?array $base, array $queryDefault
             throw new InvalidArgumentException(sprintf('Invalid "base_uri" option: host or scheme is missing in "%s".', implode('', $base)));
         }
 
+        if (null === $url['scheme'] && (null === $base || null === $base['scheme'])) {
+            throw new InvalidArgumentException('Invalid URL: scheme is missing. Did you forget to add "http(s)://"? ');
```

Comment:
extra space at the end, + this should mention the faulty URL, see message below

Reply:
I've updated the message to contain the url